transformer_engine/jax/praxis/transformer.py

```diff
@@ -59,30 +62,114 @@ def __call__(self, q_seqlen: JTensor, k_seqlen: JTensor, bidirectional: bool = T
+    qkv_layout: str = 'bshd_bshd_bshd'
+    scale_factor: float | None = None
+    transpose_batch_sequence: bool = True
+
+    def setup(self) -> None:
+        """setup"""
+        super().setup()
+
+        dpa_cls = partial(flax_DotProductAttention,
+                          head_dim=self.head_dim,
+                          num_attention_heads=self.num_attention_heads,
+                          num_gqa_groups=self.num_gqa_groups,
+                          attn_mask_type=self.attn_mask_type,
+                          attn_bias_type=self.attn_bias_type,
+                          attention_dropout=self.attention_dropout,
+                          dtype=self.dtype,
+                          dropout_rng_name=self.dropout_rng_name,
+                          float32_logits=self.float32_logits,
+                          qkv_layout=self.qkv_layout,
+                          scale_factor=self.scale_factor,
+                          transpose_batch_sequence=self.transpose_batch_sequence)
+
+        self.create_layer("dot_product_attention", dpa_cls)
+
+    def __call__(self,
+                 query: JTensor,
+                 key: JTensor,
+                 value: JTensor,
+                 mask: Optional[JTensor] = None,
+                 bias: Optional[JTensor] = None,
+                 *,
+                 deterministic: bool = False) -> JTensor:
+        """__call__"""
+        return self.dot_product_attention(query,
+                                          key,
+                                          value,
+                                          mask,
+                                          bias,
+                                          deterministic=deterministic)
+
+
 class MultiHeadAttention(TransformerEngineBaseLayer):
     """MultiHeadAttention"""
 
     head_dim: int = 64
-    num_heads: int = 16
+    num_attention_heads: int = 16
     num_gqa_groups: int | None = None
-    dropout_rate: float = 0.
+    attention_dropout: float = 0.
     dropout_rng_name: str = 'dropout'
+    input_layernorm: bool = True
     layernorm_type: str = "layernorm"
     layernorm_epsilon: float = 1e-6
     zero_centered_gamma: bool = False
+    return_layernorm_output: bool = False
     use_bias: bool = False
     bias_init: WeightInit = WeightInit.Constant(0.0)
-    apply_residual_connection_post_layernorm: bool = False
-    output_layernorm: bool = False
     attn_mask_type: str = 'causal'
-    fuse_qkv: bool = True
+    attn_bias_type: str | None = None
+    fuse_qkv_params: bool = True
     transpose_batch_sequence: bool = True
     enable_sequence_parallel: bool = False
     scale_attn_logits: bool = False
     scaled_query_init: bool = True
     float32_logits: bool = False
 
+    # Deprecated parameters
+    num_heads: int | None = None
+    dropout_rate: float | None = None
+    output_layernorm: bool | None = None
+    apply_residual_connection_post_layernorm: bool | None = None
+    fuse_qkv: bool | None = None
```

Comment:
```suggestion
    num_heads: Optional[int] = None
    dropout_rate: Optional[float] = None
    output_layernorm: Optional[bool] = None
    apply_residual_connection_post_layernorm: Optional[bool] = None
    fuse_qkv: Optional[bool] = None
```

Reply:
[Optional[int] is equalvant to int | None](https://docs.python.org/3/library/typing.html#typing.Optional), is it because for an unified coding style?

Reply:
Yes

Reply:
ok, fixed in 6f660d5a.